core/src/main/scala/scala/pickling/generator/ast.scala

```diff
@@ -0,0 +1,202 @@
+package scala.pickling
+package generator
+
+
+
+// 1. Read symbols for type
+// 2. Use an algorithm to determine the behavior of the pickle/unpickle methods (i..e generate an AST)
+//     CallConstructor(V)
+//     SetField(<date>)
+//     SetField(<id>)
+//     ---------------
+//     GetField(<date>)
+//     GetField(<id>)
+// 3. Generate code using IrAst
+
+
+/**
+ * An AST representing simple pickling behavior.  We generate this before we generate the implementing code.
+ *
+ *
+ */
+sealed trait IrAst {
```

Comment:
Not true.  it's called IrAst because it takes the scala AST and turns it into an IR then back into a scala AST.  IR means intermediate, i.e NOT an AST.   

I'm not really a fan of calling it PickleCommandTree.  We also previously had "IRs.scala" which this is meant to fully flesh out.